File: src/main/java/com/example/eventsourcing/service/command/DefaultCommandHandler.java
package com.example.eventsourcing.service.command;

import com.example.eventsourcing.domain.Aggregate;
import com.example.eventsourcing.domain.command.Command;
import org.slf4j.Logger;
import org.springframework.stereotype.Component;

@Component
public class DefaultCommandHandler implements CommandHandler<Command> {

    private static final Logger log = org.slf4j.LoggerFactory.getLogger(DefaultCommandHandler.class);

    @Override
    public void handle(Aggregate aggregate, Command command) {
        aggregate.process(command);
    }

    @Override
    public Class<Command> getCommandType() {
        return Command.class;
    }
}

File: src/main/kotlin/com/example/eventsourcing/service/command/DefaultCommandHandler.kt
package com.example.eventsourcing.service.command

import com.example.eventsourcing.domain.Aggregate
import com.example.eventsourcing.domain.command.Command
import org.slf4j.LoggerFactory

class DefaultCommandHandler : CommandHandler<Command> {
    override fun handle(aggregate: Aggregate?, command: Command?) {
        aggregate!!.process(command!!)
    }

    override val commandType = Command::class.java

    companion object {
        private val log = LoggerFactory.getLogger(DefaultCommandHandler::class.java)
    }
}
